pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator_test.go

```diff
@@ -0,0 +1,276 @@
+				gkeNetworkParams(BlueGKENetworkParamsName, BlueVPCName, BlueVPCSubnetName, []string{}),
+			},
+			interfaces: []*compute.NetworkInterface{
+				interfaces(DefaultVPCName, DefaultVPCSubnetName, "80.1.172.1", []*compute.AliasIpRange{
+					{IpCidrRange: "10.11.1.0/24", SubnetworkRangeName: DefaultSecondaryRangeA},
+				}),
+				interfaces(RedVPCName, RedVPCSubnetName, "10.1.1.1", []*compute.AliasIpRange{
+					{IpCidrRange: "172.11.1.0/24", SubnetworkRangeName: RedSecondaryRangeA},
+				}),
+			},
+			wantDefaultNwPodCIDRs: []string{"10.11.1.0/24"},
+			wantNorthInterfaces: networkv1.NorthInterfacesAnnotation{
+				{
+					Network:   RedNetworkName,
+					IpAddress: "10.1.1.1",
+				},
+			},
+			wantAdditionalNodeNetworks: networkv1.MultiNetworkAnnotation{
+				{
+					Name:  RedNetworkName,
+					Scope: "host-local",
+					Cidrs: []string{"172.11.1.0/24"},
+				},
+			},
+		},
+		{
+			desc: "interfaces without matching k8s networks should be ignored",
+			networks: []*networkv1.Network{
+				network(networkv1.DefaultNetworkName, DefaultGKENetworkParamsName),
+				network(RedNetworkName, RedGKENetworkParamsName),
+			},
+			gkeNwParams: []*networkv1alpha1.GKENetworkParamSet{
+				gkeNetworkParams(DefaultGKENetworkParamsName, DefaultVPCName, DefaultVPCSubnetName, []string{DefaultSecondaryRangeA, DefaultSecondaryRangeB}),
+				gkeNetworkParams(RedGKENetworkParamsName, RedVPCName, RedVPCSubnetName, []string{RedSecondaryRangeA, RedSecondaryRangeB}),
+			},
+			interfaces: []*compute.NetworkInterface{
+				interfaces(DefaultVPCName, DefaultVPCSubnetName, "80.1.172.1", []*compute.AliasIpRange{
+					{IpCidrRange: "10.11.1.0/24", SubnetworkRangeName: DefaultSecondaryRangeA},
+				}),
+				interfaces(RedVPCName, RedVPCSubnetName, "10.1.1.1", []*compute.AliasIpRange{
+					{IpCidrRange: "172.11.1.0/24", SubnetworkRangeName: RedSecondaryRangeA},
+				}),
+				interfaces(BlueVPCName, BlueVPCSubnetName, "84.1.2.1", []*compute.AliasIpRange{
+					{IpCidrRange: "20.28.1.0/24", SubnetworkRangeName: RedSecondaryRangeA},
+				}),
+			},
+			wantDefaultNwPodCIDRs: []string{"10.11.1.0/24"},
+			wantNorthInterfaces: networkv1.NorthInterfacesAnnotation{
+				{
+					Network:   RedNetworkName,
+					IpAddress: "10.1.1.1",
+				},
+			},
+			wantAdditionalNodeNetworks: networkv1.MultiNetworkAnnotation{
+				{
+					Name:  RedNetworkName,
+					Scope: "host-local",
+					Cidrs: []string{"172.11.1.0/24"},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			objects := make([]runtime.Object, 0)
+			for _, nw := range tc.networks {
+				objects = append(objects, nw)
+			}
+			for _, gnp := range tc.gkeNwParams {
+				objects = append(objects, gnp)
+			}
+			clientSet := fake.NewSimpleClientset(objects...)
+			ca := &cloudCIDRAllocator{
+				networkClient: clientSet,
+			}
+			gotDefaultNwCIDRs, gotNorthInterfaces, gotAdditionalNodeNetworks, _ := ca.PerformMultiNetworkCIDRAllocation(node, tc.interfaces)
```

Comment:
we should test for errors too, just add an `expectErr bool`to the `testcase` struct and here you just add
```
if expectErr && err == nil {
   t.Fatalf("expected error")
}  !expectErr && err != nil {
  t.Fatalf("unexpected error %v", err)
}
```